File: android/app/src/main/java/com/openledger/android/data/repository/AuthRepository.kt
package com.openledger.android.data.repository

import android.os.Build
import com.openledger.android.data.local.EncryptedPrefsManager
import com.openledger.android.data.remote.OpenLedgerApi
import com.openledger.android.data.remote.dto.ApiKeyCreateRequest
import com.openledger.android.data.remote.dto.LoginRequest
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AuthRepository @Inject constructor(
    private val api: OpenLedgerApi,
    private val prefs: EncryptedPrefsManager,
) {
    suspend fun checkHealth(): Result<String> = runCatching {
        val response = api.health()
        if (response.isSuccessful) response.body()!!.version
        else throw Exception("Server returned ${response.code()}")
    }

    suspend fun login(email: String, password: String): Result<String> = runCatching {
        val response = api.login(LoginRequest(email, password))
        if (!response.isSuccessful) throw Exception("Login failed: ${response.code()}")
        val token = response.body()!!.accessToken
        prefs.jwtToken = token
        token
    }

    suspend fun createApiKey(name: String): Result<String> = runCatching {
        val deviceModel = "${Build.MANUFACTURER} ${Build.MODEL}"
        val request = ApiKeyCreateRequest(
            name = name,
            platform = "android",
            deviceInfo = deviceModel,
        )
        val response = api.createApiKey(request)
        if (!response.isSuccessful) throw Exception("Failed to create API key: ${response.code()}")
        val rawKey = response.body()!!.rawKey
        prefs.apiKey = rawKey
        prefs.jwtToken = null  // No longer need JWT — API key is permanent
        rawKey
    }

    fun isConfigured(): Boolean = prefs.isConfigured()
    fun hasServerUrl(): Boolean = prefs.hasServerUrl()
    fun getServerUrl(): String? = prefs.serverUrl
    fun setServerUrl(url: String) { prefs.serverUrl = url }
    fun logout() { prefs.clear() }
}

File: ios/OpenLedger/OpenLedger/Data/Repository/AuthRepository.swift
import Foundation
import UIKit

@Observable
final class AuthRepository {

    private let api: APIClient
    private let keychain: KeychainManager

    init(api: APIClient, keychain: KeychainManager) {
        self.api = api
        self.keychain = keychain
    }

    /// Checks server health and returns the version string.
    func checkHealth() async throws -> String {
        let response: HealthResponse = try await api.request("GET", path: "/health")
        return response.version
    }

    /// Logs in with email + password, stores JWT in keychain.
    func login(email: String, password: String) async throws {
        let body = LoginRequest(email: email, password: password)
        let response: TokenResponse = try await api.request("POST", path: "/api/auth/login", body: body)
        keychain.jwtToken = response.accessToken
    }

    /// Creates an API key for this iOS device, stores the raw key, clears the JWT.
    func createApiKey(name: String) async throws {
        let deviceModel = UIDevice.current.model
        let body = ApiKeyCreateRequest(
            name: name,
            platform: "ios",
            deviceInfo: deviceModel
        )
        let response: ApiKeyCreateResponse = try await api.request("POST", path: "/api/keys/", body: body)
        keychain.apiKey = response.rawKey
        keychain.jwtToken = nil  // No longer need JWT -- API key is permanent
    }

    func setServerUrl(_ url: String) {
        keychain.serverUrl = url
    }

    func logout() {
        keychain.clear()
    }
}
